types/relay-runtime/index.d.ts

```diff
@@ -0,0 +1,1132 @@
+        interface OptimisticMutationConfig {
+            configs?: Common.RelayMutationConfig[];
+            mutation: Common.GraphQLTaggedNode;
+            variables: Common.Variables;
+            optimisticUpdater?: Common.SelectorStoreUpdater;
+            optimisticResponse?: object;
+        }
+
+        // ~~~~~~~~~~~~~~~~~~~~~
+        // fetchRelayModernQuery
+        // ~~~~~~~~~~~~~~~~~~~~~
+        // exposed through RelayModern, not Runtime directly
+        /**
+         * A helper function to fetch the results of a query. Note that results for
+         * fragment spreads are masked: fields must be explicitly listed in the query in
+         * order to be accessible in the result object.
+         *
+         * NOTE: This module is primarily intended for integrating with classic APIs.
+         * Most product code should use a Renderer or Container.
+         *
+         * TODO(t16875667): The return type should be `Promise<?SelectorData>`, but
+         * that's not really helpful as `SelectorData` is essentially just `mixed`. We
+         * can probably leverage generated flow types here to return the real expected
+         * shape.
+         */
+        function fetchRelayModernQuery(
+            environment: any, // FIXME - $FlowFixMe in facebook source code
+            taggedNode: Common.GraphQLTaggedNode,
+            variables: Common.Variables,
+            cacheConfig?: Common.CacheConfig,
+        ): Promise<any>; // FIXME - $FlowFixMe in facebook source code
+
+        // ~~~~~~~~~~~~~~~~~~~~~
+        // requestRelaySubscription
+        // ~~~~~~~~~~~~~~~~~~~~~
+        // exposed through RelayModern, not Runtime directly
+        interface GraphQLSubscriptionConfig {
+            configs?: Common.RelayMutationConfig[];
+            subscription: Common.GraphQLTaggedNode;
+            variables: Common.Variables;
+            onCompleted?(): void;
+            onError?(error: Error): void;
+            onNext?(response: object | null | undefined): void;
+            updater?(store: Common.RecordSourceSelectorProxy): void;
+        }
+        function requestRelaySubscription(
+            environment: Environment,
+            config: GraphQLSubscriptionConfig,
+        ): Common.Disposable;
+    }
+}
+
+// tslint:disable no-single-declare-module strict-export-declare-modifiers
+declare module 'relay-runtime' {
+    export import Environment = __Relay.Runtime.Environment;
+    export import Network = __Relay.Runtime.Network;
+    export import RecordSource = __Relay.Runtime.RecordSource;
+    export import Store = __Relay.Runtime.Store;
+    export import Observable = __Relay.Runtime.RelayObservable;
+    // note RecordSourceInspector is only available in dev environment
+    export import RecordSourceInspector = __Relay.Runtime.RelayRecordSourceInspector;
+    export import ConnectionHandler = __Relay.Common.Handler;
+    export import ViewerHandler = __Relay.Common.Handler;
```

Comment:
Is it possible to easily expose more/all interfaces/types that Relay would otherwise do too, instead of having to jump through hoops like these?

E.g. I can import the `RecordSourceSelectorProxy` class from the `relay-runtime` module, but I cannot import its typings from this declaration file.